wp-content/themes/pub/wporg-learn-2024/src/sensei-quiz/view.js

```diff
@@ -0,0 +1,43 @@
+/* global lesson */
+
+import { __ } from '@wordpress/i18n';
+
+function init() {
+	// Quiz page actions.
+	const actions = document.querySelector( '.sensei-quiz-actions' );
+	if ( actions ) {
+		const button = actions?.querySelector(
+			'.sensei-quiz-actions .wp-element-button.sensei-course-theme__button'
+		);
+
+		if ( ! button ) {
+			return;
+		}
+
+		if ( button.textContent.trim() === 'Pending teacher grade' ) {
+			button.innerText = __( 'Back to lesson', 'wporg-learn' );
+			button.removeAttribute( 'disabled' );
+			button.addEventListener( 'click', function () {
+				window.location.href = lesson ? lesson.link : '';
+			} );
+		}
+
+		button.style.visibility = 'visible';
+	}
+
+	// Lesson page quiz notice.
+	const noticeContent = document.querySelector( '.sensei-course-theme-lesson-quiz-notice__content' );
+	const grade = noticeContent?.querySelector( '.sensei-course-theme-lesson-quiz-notice__grade' );
+	if ( noticeContent && ! grade ) {
+		const newParagraph = document.createElement( 'p' );
+		const noticeText = noticeContent.querySelector( '.sensei-course-theme-lesson-quiz-notice__text' );
+
+		noticeText.remove();
+		newParagraph.textContent =
+			'[TBD. Sentence conveying that user is waiting for the teacher to assign a grade]';
+		newParagraph.classList.add( 'sensei-course-theme-lesson-quiz-notice__description' );
+		noticeContent.insertAdjacentElement( 'afterend', newParagraph );
+	}
```

Comment:
Have you considered doing this server side with the WP HTML API? It's rendered by the `wp-block-sensei-lms-course-theme-notices` block, and we have an example of filtering and modifying block output [here](https://github.com/WordPress/wporg-mu-plugins/blob/trunk/mu-plugins/blocks/local-navigation-bar/index.php#L14).

Reply:
We might be able to modify the pending teacher grade button to become a link button too, rather than a button with click handler.

Reply:
> Have you considered doing this server side with the WP HTML API? It's rendered by the wp-block-sensei-lms-course-theme-notices block, and we have an example of filtering and modifying block output [here](https://github.com/WordPress/wporg-mu-plugins/blob/trunk/mu-plugins/blocks/local-navigation-bar/index.php#L14).

Thanks, forget there's this function! 43705a7, 9e139b1

Reply:
> Have you considered doing this server side with the WP HTML API? It's rendered by the wp-block-sensei-lms-course-theme-notices block, and we have an example of filtering and modifying block output [here](https://github.com/WordPress/wporg-mu-plugins/blob/trunk/mu-plugins/blocks/local-navigation-bar/index.php#L14).

Thanks, forgot that there's this function! 43705a7, 9e139b1

Reply:
[Add comments and clean up conditions being used](https://github.com/WordPress/Learn/pull/2881/commits/84f8cee1287a1310272f59355b7404f4301a0a8e)
[Remove sensei-quiz block and move the logic to block-hooks](https://github.com/WordPress/Learn/pull/2881/commits/f5062c851a62e30513802720bf75fbcae0aa4c1a)